packages/xdl/src/createWebpackCompiler.js

```diff
@@ -0,0 +1,168 @@
+  }
+
+  // "invalid" event fires when you have changed a file, and Webpack is
+  // recompiling a bundle. WebpackDevServer takes care to pause serving the
+  // bundle, so if you refresh, it'll wait instead of serving the old one.
+  // "invalid" is short for "bundle invalidated", it doesn't imply any errors.
+  compiler.hooks.invalid.tap('invalid', () => {
+    if (!nonInteractive) {
+      clearConsole();
+    }
+    log(projectRoot, chalk.white('\nCompiling...'));
+  });
+
+  let isFirstCompile = true;
+
+  // "done" event fires when Webpack has finished recompiling the bundle.
+  // Whether or not you have warnings or errors, you will get this event.
+  compiler.hooks.done.tap('done', async stats => {
+    if (!nonInteractive) {
+      clearConsole();
+    }
+
+    // We have switched off the default Webpack output in WebpackDevServer
+    // options so we are going to "massage" the warnings and errors and present
+    // them in a readable focused way.
+    // We only construct the warnings and errors for speed:
+    // https://github.com/facebook/create-react-app/issues/4492#issuecomment-421959548
+    const statsData = stats.toJson({
+      all: false,
+      warnings: true,
+      errors: true,
+    });
+
+    const messages = formatWebpackMessages(statsData);
+
+    if (messages.errors.length > 0) {
+      devSocket.errors(messages.errors);
+    } else if (messages.warnings.length > 0) {
+      devSocket.warnings(messages.warnings);
+    }
+
+    // New line after the bundle analyzer finishes
+    log(projectRoot, ` `);
+
+    const isSuccessful = !messages.errors.length && !messages.warnings.length;
+    if (isSuccessful) {
+      log(projectRoot, chalk.bold.cyan(`Compiled successfully!`));
+      printPreviewNotice(projectRoot);
+    }
+    if (isSuccessful && (!nonInteractive || isFirstCompile)) {
+      printInstructions(projectRoot, appName, urls, useYarn);
+    }
+    onFinished();
+    isFirstCompile = false;
+
+    // If errors exist, only show errors.
+    if (messages.errors.length) {
+      // Only keep the first error. Others are often indicative
+      // of the same problem, but confuse the reader with noise.
+      if (messages.errors.length > 1) {
+        messages.errors.length = 1;
+      }
+      logError(projectRoot, chalk.red('Failed to compile.\n'));
+      logError(projectRoot, messages.errors.join('\n\n'));
+      return;
+    }
+
+    // Show warnings if no errors were found.
+    if (messages.warnings.length) {
+      logWarning(projectRoot, chalk.yellow('Compiled with warnings.\n'));
+      logWarning(projectRoot, messages.warnings.join('\n\n'));
+
+      // Teach some ESLint tricks.
+      logWarning(
+        projectRoot,
+        '\nSearch for the ' +
+          chalk.underline(chalk.yellow('keywords')) +
+          ' to learn more about each warning.'
+      );
+      logWarning(
+        projectRoot,
+        'To ignore, add ' + chalk.cyan('// eslint-disable-next-line') + ' to the line before.\n'
```

Comment:
These instructions seem wrong: I don't think we run the code through ESLint, so `// eslint-disable-next-line` should not have any effect. Also not sure what "keywords" refers to.

<img width="889" alt="Screenshot 2019-04-11 at 16 00 13" src="https://user-images.githubusercontent.com/497214/55959171-ee90b980-5c72-11e9-8e3b-dc0b11072410.png">
Got these warnings when running in `native-component-list`.